beacon_node/eth2_libp2p/src/rpc/codec/ssz_snappy.rs

```diff
@@ -246,6 +293,14 @@ impl<TSpec: EthSpec> Decoder for SSZSnappyOutboundCodec<TSpec> {
     type Error = RPCError;
 
     fn decode(&mut self, src: &mut BytesMut) -> Result<Option<Self::Item>, Self::Error> {
+        // Read the context bytes if required
+        if self.protocol.version == Version::V2 && self.context_bytes.is_none() {
+            let context_bytes = src.split_to(4);
```

Comment:
This is very dangerous. It will panic if src is < 4 bytes. Are we 100% sure src must be at least 4 bytes long here?

Reply:
Thank you! will fix this.